### PR TITLE
V2.0 : Created GetMovementServlet and GetMovementHashtagsServlet, Added correct URL for CreateMovementServlet

### PR DESCRIPTION
**Pull request by rashelrr**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 1]
```diff
@@ -12,24 +12,37 @@
...
-    String hashTag = Jsoup.clean(request.getParameter("hashTag"), Whitelist.none());
-    
+    String hashtag = Jsoup.clean(request.getParameter("hashtag"), Whitelist.none());
+    String video = Jsoup.clean(request.getParameter("video"), Whitelist.none());
+    String image = Jsoup.clean(request.getParameter("image"), Whitelist.none());
+    String resources = Jsoup.clean(request.getParameter("resources"), Whitelist.none());
     
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
-    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Task");
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("MovementByUser");
```
Comment (enriquejosepadilla) [inline src/main/java/com/google/sps/servlets/FormHandlerServlet.java:32]: "Movement"

File: src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 2]
```diff
@@ -12,24 +12,37 @@
...
-  } */
\ No newline at end of file
+    FullEntity movementEntity =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("name", name)
+            .set("description", description)
+            .set("hashtag", hashtag)
+            .set("video", video)
+            .set("image", image)
+            .set("resources", resources)
```
Comment (enriquejosepadilla) [inline src/main/java/com/google/sps/servlets/FormHandlerServlet.java:41]: donationLink

**Review by enriquejosepadilla — CHANGES_REQUESTED** [review]

**Comment by enriquejosepadilla** [comment]
fix merge conflicts please

File: src/main/java/com/google/sps/servlets/CreateMovementServlet.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            .set("hashtag", hashtag)
+            .set("video", video)
+            .set("image", image)
+            .set("donation", donation) 
+            .set("resources", resources) 
+            .build();
+    datastore.put(movementEntity);
+
+    // response.sendRedirect("/index.html");
+    response.sendRedirect("https://spring21-sps-1.uc.r.appspot.com/movement?=" + hashtag);
```
Comment (enriquejosepadilla) [inline src/main/java/com/google/sps/servlets/CreateMovementServlet.java:51]: does this never change?

**Review by enriquejosepadilla — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/CreateMovementServlet.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    String description = Jsoup.clean(request.getParameter("description"), Whitelist.none());
+    String hashtag = Jsoup.clean(request.getParameter("hashtag"), Whitelist.none());
+    String video = Jsoup.clean(request.getParameter("video"), Whitelist.none());
+    String image = Jsoup.clean(request.getParameter("image"), Whitelist.none());
+    String donation = Jsoup.clean(request.getParameter("donation"), Whitelist.none());
+    String resources = Jsoup.clean(request.getParameter("resources"), Whitelist.none());
+    
+    // removes '#'
+    if (!hashtag.isEmpty())
+      hashtag = hashtag.substring(1);
```
Comment (enriquejosepadilla) [inline src/main/java/com/google/sps/servlets/CreateMovementServlet.java:36]: check that the first sign is # before you remove it. If it isnt, dont remove anything

File: src/main/java/com/google/sps/servlets/CreateMovementServlet.java (hunk 3 above)
Comment (rashelrr) [reply src/main/java/com/google/sps/servlets/CreateMovementServlet.java:51]: I'm not sure and I don't know how to verify this. When I press submit on the form page, the following url is loaded: https://spring21-sps-1.uc.r.appspot.com/form-handler. Not sure if it should be this then?

**Review by enriquejosepadilla — APPROVED** [review]

**Pull request merged**